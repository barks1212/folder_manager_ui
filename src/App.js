import React from 'react';
import logo from './logo.svg';

class App extends React.Component {
    render() {
        return (
        <div>test</div>
        )
    }
}


export default App;