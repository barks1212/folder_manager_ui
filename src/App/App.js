import React from 'react';

import Layout from '../HOC/Layout/Layout';
import FolderList from '../containers/FolderList/FolderList';

class App extends React.Component {
    render() {
        return (
            <Layout>
                <FolderList />
            </Layout>
        )
    }
}


export default App;