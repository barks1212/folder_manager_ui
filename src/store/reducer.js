import folderData from '../data/folders.json';
import * as actionTypes from './actions';

const initialState = {
    folderData,
    sort: null,
}

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case actionTypes.SET_SORT:
            if (action.sort === 'name') {
                return {
                    ...state,
                    sort: 'name'
                }
            }
            if (action.sort === 'date') {
                return {
                    ...state,
                    sort: 'date'
                };
            };
            break;
        case actionTypes.SORT_FILES:
            let sortedState;
            if (action.sort === 'name') {
                sortedState = folderData.sort((a, b) => {
                    if (a.name < b.name) return -1;
                    if (a.name > b.name) return 1;
                    return 0;
                });
                return {
                    ...state,
                    folderData: sortedState
                };
            };
            break;
        default:
            return state;
    };
};

export default reducer;