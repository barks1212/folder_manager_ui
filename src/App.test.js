import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';
import {createStore} from 'redux';
import {Provider} from 'react-redux';

import reducer from './store/reducer';
import App from './App';

afterEach(cleanup);

const store = createStore(reducer);

describe('App', () => {
    it('should render', () => {
        const {asFragment} = render(
        <Provider store={store}>
            <App />
        </Provider>);
        expect(asFragment()).toMatchSnapshot();
    });
});