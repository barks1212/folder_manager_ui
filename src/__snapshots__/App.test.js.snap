// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<DocumentFragment>
  <header
    class="Toolbar"
  >
    <select
      class="Select"
      placeholder="sort by"
    >
      <option
        disabled=""
        selected=""
        value="Sort By"
      >
        Sort by
      </option>
      <option
        value="name"
      >
        Name
      </option>
      <option
        value="added"
      >
        Date
      </option>
    </select>
  </header>
  <main
    class="Content"
  >
    <div
      class="FolderList"
    >
      <div
        data-testid="file"
      >
        <div
          class="File"
        >
          <ul>
            <li>
              pdf
            </li>
            <li
              class="Name"
            >
              Employee Handbook
            </li>
            <li>
              2017-01-06
            </li>
          </ul>
        </div>
      </div>
      <div
        data-testid="file"
      >
        <div
          class="File"
        >
          <ul>
            <li>
              pdf
            </li>
            <li
              class="Name"
            >
              Public Holiday policy
            </li>
            <li>
              2016-12-06
            </li>
          </ul>
        </div>
      </div>
      <div
        data-testid="folder"
      >
        <div
          class="Folder"
          data-testid="folderWrapper"
        >
          <ul>
            <li
              class="Name"
            >
              Expenses
            </li>
            <li>
              <i
                class="fa fa-chevron-down Chev"
              />
            </li>
          </ul>
        </div>
        <div
          data-testid="subFiles"
        />
      </div>
      <div
        data-testid="file"
      >
        <div
          class="File"
        >
          <ul>
            <li>
              csv
            </li>
            <li
              class="Name"
            >
              Cost centres
            </li>
            <li>
              2016-08-12
            </li>
          </ul>
        </div>
      </div>
      <div
        data-testid="folder"
      >
        <div
          class="Folder"
          data-testid="folderWrapper"
        >
          <ul>
            <li
              class="Name"
            >
              Misc
            </li>
            <li>
              <i
                class="fa fa-chevron-down Chev"
              />
            </li>
          </ul>
        </div>
        <div
          data-testid="subFiles"
        />
      </div>
    </div>
  </main>
</DocumentFragment>
`;
